pkg/plugins/runtime/k8s/controllers/pod_controller.go

```diff
@@ -541,3 +546,63 @@ func EndpointSliceToPodsMapper(l logr.Logger, client kube_client.Client) kube_ha
 		return req
 	}
 }
+
+func MeshToPodsMapper(l logr.Logger, client kube_client.Client) kube_handler.MapFunc {
+	l = l.WithName("mesh-to-pods-mapper")
+	return func(ctx context.Context, obj kube_client.Object) []kube_reconcile.Request {
+		mesh := obj.(*mesh_k8s.Mesh)
+
+		l := l.WithValues(
+			"Mesh",
+			kube_types.NamespacedName{Namespace: mesh.Namespace, Name: mesh.Name},
+		)
+
+		pods := &kube_core.PodList{}
+		if err := client.List(ctx, pods, kube_client.InNamespace(obj.GetNamespace()), kube_client.MatchingLabels(map[string]string{
+			mesh_proto.MeshTag: mesh.Name,
+		})); err != nil {
+			l.Error(err, "failed to fetch Pods matching for mesh")
+			return nil
+		}
+
+		var req []kube_reconcile.Request
+		for _, pod := range pods.Items {
+			req = append(req, kube_reconcile.Request{
+				NamespacedName: kube_types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name},
+			})
+		}
+		return req
+	}
+}
+
+type MeshServiceExclusivePredicate struct {
```

Comment:
have you checked if changing something else on the mesh (like mtls) doesn't reconcile pods?